pkg/controllers/util/federatedclient/podinformer.go

```diff
@@ -0,0 +1,109 @@
+package federatedclient
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/informers"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (f *federatedClientFactory) addPodInformer(informer informers.SharedInformerFactory, client kubeclient.Interface) {
+	informer.InformerFor(&corev1.Pod{}, func(k kubeclient.Interface, duration time.Duration) cache.SharedIndexInformer {
+		return cache.NewSharedIndexInformer(
+			f.podListerWatcher(client),
+			&corev1.Pod{},
+			0,
+			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		)
+	})
+}
+
+func (f *federatedClientFactory) podListerWatcher(client kubeclient.Interface) cache.ListerWatcher {
+	return &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			cur := atomic.LoadInt32(&f.availablePodListers)
+			if cur == 0 {
+				return nil, fmt.Errorf("do not have enought lister")
+			}
+			if cur > 0 {
+				if !atomic.CompareAndSwapInt32(&f.availablePodListers, cur, cur-1) {
+					return nil, fmt.Errorf("failed to compete for lister")
+				}
+				defer atomic.AddInt32(&f.availablePodListers, 1)
+			}
+			pods, err := client.CoreV1().Pods(corev1.NamespaceAll).List(context.TODO(), options)
+			if err != nil {
+				return nil, err
+			}
+			for i, pod := range pods.Items {
+				pods.Items[i] = simplifyPod(pod)
+			}
+			return pods, nil
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			watcher, err := client.CoreV1().Pods(corev1.NamespaceAll).Watch(context.TODO(), options)
+			if err != nil {
+				return nil, err
+			}
+
+			// It's easy for a consumer to add buffering via an extra
+			// goroutine/channel, but impossible for them to remove it,
+			// so nonbuffered is better. -- from watch.NewStreamWatcher
+			proxyCh := make(chan watch.Event)
```

Comment:
It's simpler to just specify a `Transform` function (i.e.`simplifyPod`) and return `watcher` as is. Do you think this approach might work?

Reply:
For LIST, it might be necessary to explicitly simplify the pods before returning to (hopefully) expedite GC, but WATCH should not have the same concern. @SOF3 WDYT?

Reply:
i think both approaches are fine. but since we are already transforming in the list function, maybe we can forgo using the `Transform` function all together to save some cpu cycles?